packit_service/worker/handlers/abstract.py

```diff
@@ -161,12 +174,47 @@ class JobHandler(Handler):
     triggers: List[TheJobTriggerType]
 
     def __init__(
-        self, config: ServiceConfig, job_config: Optional[JobConfig], event: Event
+        self,
+        package_config: PackageConfig,
+        job_config: Optional[JobConfig],
+        event: dict,
     ):
-        super().__init__(config)
+        self.package_config: PackageConfig = package_config
         self.job_config: Optional[JobConfig] = job_config
+        self.event_type: str = event.get("event_type")
+        self.trigger: TheJobTriggerType = TheJobTriggerType(event.get("trigger"))
+        self.user_login: str = event.get("user_login")
+        self.trigger_id: int = event.get("trigger_id")
+        self.project_url: str = event.get("project_url")
+
         self.event = event
+        self._db_trigger: Optional[AbstractTriggerDbType] = None
+        self._project: Optional[GitProject] = None
+        self._config: Optional[ServiceConfig] = None
         self._clean_workplace()
 
+    @property
+    def db_trigger(self):
+        if not self._db_trigger and self.trigger_id is not None:
+            if self.trigger in (
+                TheJobTriggerType.pull_request,
+                TheJobTriggerType.pr_comment,
+                TheJobTriggerType.pr_label,
+            ):
+                self._db_trigger = PullRequestModel.get_by_id(self.trigger_id)
+            elif self.trigger == TheJobTriggerType.issue_comment:
+                self._db_trigger = IssueModel.get_by_id(self.trigger_id)
+            elif self.trigger == TheJobTriggerType.release:
+                self._db_trigger = ProjectReleaseModel.get_by_id(self.trigger_id)
+            elif self.trigger in TheJobTriggerType.push:
+                self._db_trigger = GitBranchModel.get_by_id(self.trigger_id)
+        return self._db_trigger
```

Comment:
I was a bit lazy last time when I create a similar if-else but we should share this code across handlers.
(Later if you don't want to do it (here).)

Same for other trigger-type-related properties.